ngraph_creator/operations/src/Channel_Shuffle.cpp

```diff
@@ -0,0 +1,53 @@
+#include <Channel_Shuffle.hpp>
+#define LOG_TAG "Channel_Shuffle"
+
+namespace android {
+namespace hardware {
+namespace neuralnetworks {
+namespace nnhal {
+
+Channel_Shuffle::Channel_Shuffle(int operationIndex) : OperationsBase(operationIndex) {
+    mDefaultOutputIndex = sModelInfo->getOperationOutput(mNnapiOperationIndex, 0);
+}
+
+bool Channel_Shuffle::validate() {
+    // check output type
+    if (!checkOutputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkOutputOperandType(0, (int32_t)OperandType::TENSOR_QUANT8_ASYMM)) {
+        return false;
+    }
+
+    // Check all input types
+    if (!checkInputOperandType(0, (int32_t)OperandType::TENSOR_FLOAT32) &&
+        !checkInputOperandType(0, (int32_t)OperandType::TENSOR_QUANT8_ASYMM)) {
+        return false;
+    }
+    if (!checkInputOperandType(1, (int32_t)OperandType::INT32)) {
+        return false;
+    }
+    if (!checkInputOperandType(2, (int32_t)OperandType::INT32)) {
+        return false;
+    }
+
+    // Check input rank
+    const auto inputRank = getInputOperandDimensions(0).size();
+    if (inputRank != 4) return false;
```

Comment:
Since a lot of test cases get ignored with this check, can you check if you can add dims of 1 where necessary to make the tensor 4 dims. The axis value needs to be changed accordingly.

Reply:
Updated with check for axis range and conversion of input node to 4D to resolve fatal CTS issues.